opendut-types/src/peer/executor.rs

```diff
@@ -33,6 +34,13 @@ pub enum Engine {
     Podman
 }
 
+impl Engine {
+    pub fn iterator() -> Iter<'static, Engine> {
```

Comment:
This can be solved by the [strum](https://crates.io/crates/strum) crate. strum provides a derive macro call [`EnumIter`](https://docs.rs/strum_macros/latest/strum_macros/derive.EnumIter.html) which generates a new type to iterate over the variants of an enum. The great advantage is, that you don't have to manually maintain the implementation of the iterator.